build-system/tasks/helpers.js

```diff
@@ -522,6 +522,68 @@ async function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
   }
 }
 
+/**
+ * Transforms a given JavaScript file entry point with esbuild and babel, and
+ * watches it for changes (if required).
+ * Used by 3p iframe vendors.
+ * @param {string} srcDir
+ * @param {string} srcFilename
+ * @param {string} destDir
+ * @param {?Object} options
+ * @return {!Promise}
+ */
+async function compileJsWithEsbuild(srcDir, srcFilename, destDir, options) {
+  const startTime = Date.now();
+  const entryPoint = path.join(srcDir, srcFilename);
+  const destFilename = maybeToEsmName(
+    options.minify ? options.minifiedName : options.toName
+  );
+  const destFile = path.join(destDir, destFilename);
+
+  if (watchedTargets.has(entryPoint)) {
+    return watchedTargets.get(entryPoint).rebuild();
+  }
+
+  const plugin = getEsbuildBabelPlugin(
+    options.minify ? 'minified' : 'unminified',
+    /* enableCache */ true
+  );
+  const buildResult = await esbuild
+    .build({
+      entryPoints: [entryPoint],
+      bundle: true,
+      sourcemap: true,
+      outfile: destFile,
+      plugins: [plugin],
+      minify: options.minify,
+      incremental: !!options.watch,
+      logLevel: 'silent',
+    })
+    .then((result) => {
+      finishBundle(srcFilename, destDir, destFilename, options, startTime);
+      return result;
+    })
+    .catch((err) => handleBundleError(err, !!options.watch, destFilename));
+
+  if (options.watch) {
+    watchedTargets.set(entryPoint, {
+      rebuild: async () => {
+        const time = Date.now();
+        const buildPromise = buildResult
+          .rebuild()
+          .then(() =>
+            finishBundle(srcFilename, destDir, destFilename, options, time)
```

Comment:
Until now, `finishBundle` was only invoked during unminified builds. It needs to be changed to account for the fact that it could also be run on minified binaries.

Here's an updated implementation for the function:

```js
/**
 * Performs the final steps after a JS file is bundled and optionally minified
 * with esbuild and babel.
 * @param {string} srcFilename
 * @param {string} destDir
 * @param {string} destFilename
 * @param {?Object} options
 * @param {number} startTime
 */
async function finishBundle(
  srcFilename,
  destDir,
  destFilename,
  options,
  startTime
) {
  combineWithCompiledFile(
    srcFilename,
    path.join(destDir, destFilename),
    options
  );

  const logPrefix = options.minify ? 'Minified' : 'Compiled';
  let {latestName} = options;
  if (latestName) {
    if (!options.minify) {
      latestName = latestName.replace(/\.js$/, '.max.js');
    }
    fs.copySync(
      path.join(destDir, options.toName),
      path.join(destDir, latestName)
    );
    endBuildStep(logPrefix, `${destFilename} → ${latestName}`, startTime);
  } else {
    endBuildStep(logPrefix, destFilename, startTime);
  }

  const targets = options.minify ? MINIFIED_TARGETS : UNMINIFIED_TARGETS;
  const target = path.basename(destFilename, path.extname(destFilename));
  if (targets.includes(target)) {
    await applyAmpConfig(
      path.join(destDir, destFilename),
      /* localDev */ true,
      /* fortesting */ options.fortesting
    );
  }
}
```

Reply:
Thanks you! Adopted.